contracts/MeshCrowdsale.sol

```diff
@@ -2,30 +2,90 @@ pragma solidity ^0.4.15;
 
 import './MeshToken.sol';
 import 'zeppelin-solidity/contracts/crowdsale/CappedCrowdsale.sol';
+import 'zeppelin-solidity/contracts/math/SafeMath.sol';
+import 'zeppelin-solidity/contracts/ownership/Ownable.sol';
 
 
-contract MeshCrowdsale is CappedCrowdsale {
+contract MeshCrowdsale is CappedCrowdsale, Ownable {
+
+  using SafeMath for uint256;
 
   /**
-   * @dev address of token that is being minted by the crowdsale
-   * replace 0x00 with the token address when deploying
+   * @dev weiLimits keeps track of amount of wei that can be contibuted by an address.
    */
-  address public tokenAddress = address(0x00);
+  mapping (address => uint256) public weiLimits;
+
+  /**
+   * @dev weiContributions keeps track of amount of wei that are contibuted by an address.
+   */
+  mapping (address => uint256) public weiContributions;
+
+  /*---------------------------------constructor---------------------------------*/
 
   /**
    * @dev Constructor for MeshCrowdsale contract
    */
-  function MeshCrowdsale(uint256 _startTime, uint256 _endTime, uint256 _rate, address _wallet, uint256 _cap)
-    CappedCrowdsale(_cap)
-    Crowdsale(_startTime, _endTime, _rate, _wallet)
-    public
-    {
-    }
+  function MeshCrowdsale(uint256 _startTime, uint256 _endTime, uint256 _rate, address _wallet, uint256 _cap, address _tokenAddress)
+  CappedCrowdsale(_cap)
+  Crowdsale(_startTime, _endTime, _rate, _wallet)
+  public
+  {
+    token = MeshToken(_tokenAddress);
+  }
+
+  /*---------------------------------overridden methods---------------------------------*/
 
-  // creates the token to be sold.
+  // dummy function to create a token instance
   // overridden method to point to already deployed token
+  /**
+   * overriding Crowdsale#createTokenContract to point to a 0 address
+   * actual token address mapping happens in the Constructor.
```

Comment:
Can we be more specific and specify that the actual token address initialization is done in the MeshCrowdsale constructor?